generator/default.p9k

```diff
@@ -366,15 +367,28 @@ function __p9k_build_segment_cache() {
 
 ###############################################################
 # @description
-#   Refresh a single item in the cache
+#   Refresh a single item/segment in the cache
 ##
 # @args
-#   $1 string The serialized segment data
+#   $1 segment name
+#   $2 boolean true to trigger prompt rendering
 ##
 function __p9k_refresh_cache_item() {
-  local -a segment_meta=("${(@s:·|·:)1}")
+  local segment="$1"
+  for alignment in left right; do
+    for index in $(p9k::find_in_array "${segment}" "${__P9K_DATA[${alignment}_segments]}"); do
+      local joined=false
+      local segmentMetaVar="P9K_${(U)alignment}_PROMPT_ELEMENTS[${index}]"
+      p9k::segment_is_tagged_as "joined" ${${(P)segmentMetaVar}%%::*} && joined=true
 
-  __p9k_segment_cache["${segment_meta[2]}::${segment_meta[3]}"]="${1}"
+      local cache_key="${alignment}::${segment}"
+
+      prompt_${segment} "${alignment}" "$index" "${joined}"
+      __p9k_segment_cache["${cache_key}"]="${__P9K_DATA[SEGMENT_RESULT]}"
+    done
+  done
+
+  [[ $2 == "true" ]] && __p9k_render "true"
```

Comment:
I didn't add this in the first place because for separation of concerns. Redraw the prompt is a different thing than update the cache..